tests/compute/credentials_test.go

```diff
@@ -0,0 +1,320 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package compute
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+
+	expect "github.com/google/goexpect"
+
+	kubev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+var _ = SIGDescribe("Guest Access Credentials", func() {
+
+	const (
+		fedoraRunningTimeout    = 120
+		guestAgentConnetTimeout = 2 * time.Minute
+		denyListTimeout         = 2 * time.Minute
+		fedoraPassword          = "fedora"
+	)
+
+	Context("with qemu guest agent", func() {
+		withSSHPK := func(secretName string) libvmi.Option {
+			return func(vmi *v1.VirtualMachineInstance) {
+				vmi.Spec.AccessCredentials = append(vmi.Spec.AccessCredentials,
+					v1.AccessCredential{
+						SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
+							Source: v1.SSHPublicKeyAccessCredentialSource{
+								Secret: &v1.AccessCredentialSecretSource{
+									SecretName: secretName,
+								},
+							},
+							PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
+								QemuGuestAgent: &v1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
+									Users: []string{"fedora"},
+								},
+							},
+						}})
+			}
+		}
+
+		withPassword := func(secretName string) libvmi.Option {
+			return func(vmi *v1.VirtualMachineInstance) {
+				vmi.Spec.AccessCredentials = append(vmi.Spec.AccessCredentials,
+					v1.AccessCredential{
+						UserPassword: &v1.UserPasswordAccessCredential{
+							Source: v1.UserPasswordAccessCredentialSource{
+								Secret: &v1.AccessCredentialSecretSource{
+									SecretName: secretName,
+								},
+							},
+							PropagationMethod: v1.UserPasswordAccessCredentialPropagationMethod{
+								QemuGuestAgent: &v1.QemuGuestAgentUserPasswordAccessCredentialPropagation{},
+							},
+						},
+					})
+			}
+		}
```

Comment:
This is only used once, but together with the previous one could also let me think about a new separate file under `libvmi`. WDYT?

Reply:
Answered above.